### PR TITLE
aws: Update for new install data model

### PR DESCRIPTION
**Pull request by nbrahms**
P0's data model for AWS installations has changed. Update the CLI to work with this new data model.

### CODE REVIEW

File: src/commands/__tests__/ssh.test.ts [hunk 1]
```diff
@@ -32,9 +32,9 @@ mockGetDoc({
   workflows: {
     items: [
       {
-        identifier: "test-account",
```
Comment (GGonryun) [inline src/commands/__tests__/ssh.test.ts:35]: 🤔

File: src/commands/aws/role.ts [hunk 2]
```diff
@@ -46,10 +46,10 @@ export const role = (yargs: yargs.Argv<{ account: string | undefined }>) =>
...
   );
 
-const isOktaSamlConfig = (
-  config: AwsItemConfig
-): config is AwsItemConfig & { uidLocation: AwsOktaSamlUidLocation } =>
-  config.uidLocation?.id === "okta_saml_sso";
+const isFederatedLogin = (
+  config: AwsItem
+): config is AwsItem & { login: AwsFederatedLogin } =>
+  config.login?.type === "federated";
```
Comment (gergas3) [inline src/commands/aws/role.ts:52]: Should this be the equivalent stricter check that verifies all the way down to `login.provider.method == saml` (and `okta`, etc.)?

File: src/plugins/aws/config.ts [hunk 3]
```diff
@@ -23,11 +24,19 @@ export const getAwsConfig = async (
...
+    ),
+    ([id]) => id
+  );
   const item = account
-    ? config?.workflows?.items.find(
-        (i) => i.state === "installed" && i.account.id === account
-      )
-    : config?.workflows?.items[0];
+    ? allItems.find(([id, { label }]) => id === account || label === account)
+    : allItems.length !== 1
```
Comment (gergas3) [inline src/plugins/aws/config.ts:35]: Can this ever be more than 1? `account` or `label` would have to match multiple items.
Suggestion: just pick the first one like we used to.

**Review by gergas3 — APPROVED** [review]

**Review by GGonryun — APPROVED** [review]
👍

File: src/commands/aws/role.ts (hunk 2 above)
Comment (nbrahms) [reply src/commands/aws/role.ts:52]: not necessary I think; when we add more complications we can update this

File: src/plugins/aws/config.ts (hunk 3 above)
Comment (nbrahms) [reply src/plugins/aws/config.ts:35]: it doesn't make any sense to just pick the first one though; I discovered this by trying to `p0 aws role ls`, which would randomly fail or pass depending on wether the first was the one that was federated login

File: src/commands/__tests__/ssh.test.ts (hunk 1 above)
Comment (nbrahms) [reply src/commands/__tests__/ssh.test.ts:35]: yeah, debugging artifact 🤷

**Pull request merged**